morecantile/models.py

```diff
@@ -93,22 +104,18 @@ class CRSWKT(BaseModel):
 #     )
 
 
-class CRSType(BaseModel):
+class CRSType(RootModel[Union[str, Union[CRSUri, CRSWKT]]]):
     """CRS model.
 
     Ref: https://github.com/opengeospatial/ogcapi-tiles/blob/master/openapi/schemas/common-geodata/crs.yaml
 
     Code generated using https://github.com/koxudaxi/datamodel-code-generator/
     """
 
-    __root__: Union[str, Union[CRSUri, CRSWKT]] = Field(..., title="CRS")
-    _pyproj_crs: CRS = PrivateAttr()
-
-    def __init__(self, **data):
-        """Custom Init to validate CRS string and create Pyproj CRS object."""
-        super().__init__(**data)
-
-        self._pyproj_crs = CRS.from_user_input(data.get("__root__"))
+    @property
+    def _pyproj_crs(self) -> CRS:
+        """return pyproj CRS."""
+        return CRS.from_user_input(self.root)
```

Comment:
Sadly it seems that we can't add private attributes to Custom RootModel 🤷 

might need to open an issue in pydantic about this